File: 백준/Silver/1926. 그림/그림.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "JS/example.txt";

const input = require("fs")
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.split(" ").map(Number));

const [N, M] = input.shift();
const dx = [1, -1, 0, 0];
const dy = [0, 0, -1, 1];
const result = [];

function bfs(a, b) {
  const q = [];
  q.push([a, b]);
  let count = 1;
  input[a][b] = 0;

  while (q.length > 0) {
    const [x, y] = q.shift();

    for (let i = 0; i < 4; i++) {
      let nx = x + dx[i];
      let ny = y + dy[i];

      if (0 <= nx && nx < N && 0 <= ny && ny < M && input[nx][ny] === 1) {
        input[nx][ny] = 0;
        count++;
        q.push([nx, ny]);
      }
    }
  }
  return count;
}

for (let i = 0; i < N; i++) {
  for (let j = 0; j < M; j++) {
    if (input[i][j] === 1) {
      result.push(bfs(i, j));
    }
  }
}
console.log(result.length);
if (result.length === 0) {
  console.log(0);
} else {
  console.log(Math.max(...result));
}

File: 백준/Silver/1926. 그림/그림.py
import sys
from collections import deque

def bfs(a,b):
  tmp = 1
  q = deque()
  q.append((a,b))
  graph[a][b] = 0

  while q:
    y, x = q.popleft()

    for i in range(4):
      nx = x + dx[i]
      ny = y + dy[i]
      if 0 <= nx < m and 0 <= ny < n:
        if graph[ny][nx]:
          q.append((ny,nx))
          graph[ny][nx] = 0
          tmp += 1
  return tmp

n,m = map(int, sys.stdin.readline().split())

graph = [list(map(int,sys.stdin.readline().split())) for _ in range(n)]

dx = [0,0,-1,1]
dy = [-1,1,0,0]

result = 0
count = 0

for i in range(n):
  for j in range(m):
    if graph[i][j]:
      count += 1
      result = max(result, bfs(i,j))

print(count)
print(result)